pkg/reconciler/ingress/config/istio.go

```diff
@@ -43,24 +44,36 @@ const (
 	ReconcileExternalGatewayKey = "reconcileExternalGateway"
 )
 
-var (
-	defaultGateway = Gateway{
-		GatewayName: "knative-ingress-gateway",
+func defaultGateways() []Gateway {
+	return []Gateway{{
+		Namespace: system.Namespace(),
+		Name:      "knative-ingress-gateway",
 		ServiceURL: fmt.Sprintf("istio-ingressgateway.istio-system.svc.%s",
 			network.GetClusterDomainName()),
-	}
-	defaultLocalGateway = Gateway{
-		GatewayName: "cluster-local-gateway",
+	}}
+}
+
+func defaultLocalGateways() []Gateway {
```

Comment:
nit: function here seems not necessary.
could we just use:
```
var defaultLocalGateways = []Gateway{{..}}
```

Reply:
that doesn't work because I used `system.Namespace()` which needs to be called either inside K8s environment or test.